execinfo.h isn't available <10.5, so make sure it is only included when
actually necessary

--- ld/ld.cpp
+++ ld/ld.cpp
@@ -37,7 +37,9 @@
 #include <errno.h>
 #include <limits.h>
 #include <unistd.h>
+#ifndef NDEBUG
 #include <execinfo.h>
+#endif
 #include <mach/mach_time.h>
 #include <mach/vm_statistics.h>
 #include <mach/mach_init.h>
